src/Core/Domain/Category/QueryResult/CategoryForTree.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Category\QueryResult;
+
+class CategoryForTree
+{
+    /**
+     * @var int
+     */
+    private $categoryId;
+
+    /**
+     * @var string
+     */
+    private $categoryName;
+
+    /**
+     * @var CategoryForTree[]
+     */
+    private $childCategories;
```

Comment:
```suggestion
    private $childrenCategories;
```

or simply

```suggestion
    private $children;
```

since the categories suffix is a bit redundant

Reply:
i don't find it so redundant here, I think its clear (we've had much more redundant vars in product migration).
$childCategories helps to state the fact that its multiple categories, whereas "child|children" might sound as single one too. So im not up for this change, i like my version :stuck_out_tongue:

Reply:
Ok for the suffix then I agree it's acceptable However since it's plural I'd still use children so I would use `childrenCategories`